test/fail_compilation/retscope.d

```diff
@@ -250,3 +250,20 @@ void escape4() @safe
     scope FunDG g = ()        { return &x; };
 }
 
+/**************************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope.d(267): Error: cannot take address of scope local p in @safe function escape5
+---
+*/
+
+void escape5() @safe
+{
+    int* q;
+    scope int* p;
+    scope int** pp = &q; // ok
+    pp = &p; // error
```

Comment:
The lifetime of pp is shorter than the lifetime of p, so shouldn't this be allowed?
We could go with a conservative implemenation for now.